### ISSUE TITLE
TextAndImageButton displayed incorrectly

### ISSUE THREAD

**Issue by rko281** [dolphinsmalltalk/Dolphin#1032]
There is insufficient gap between the image and text, such that the two overlap - example:
```
| button |
button := TextAndImageButton new.
ShellView show addSubView: button.
button image: Icon warning; text: 'TEST'
```
Previously displayed as: 
![image](https://user-images.githubusercontent.com/16988710/95472212-aa22ce80-097a-11eb-81e3-c6527b592dd3.png)

Now displays as:
![image](https://user-images.githubusercontent.com/16988710/95472185-a2fbc080-097a-11eb-8237-26e5638caa92.png)

Possibly introduced by 1367d25c681e77f2f2fa33d581599079bc9e1e16

**Comment by blairmcg** [comment]
Yes, unfortunately it does break with the changes from #1028 - afterwards the text is always drawn in the same place and doesn't move with the button when the alignment is changed.

**Comment by blairmcg** [comment]
The bug is in `TextAndImageButton>>textPosOn:`. It should be adjusting the xPos with the text offset from the image, not using that as the xPos. 
The PR included more changes than I'd anticipated as it had the same title as the previous one. I'm afraid I only looked at the test, assuming the rest was the same. If I'd looked at the changes more closely I would have realised the test was  inadequate for that level of change. As I didn't look at it carefully enough myself, I've created and pushed a fix for it.
